modules/aggregations/src/test/java/org/elasticsearch/aggregations/pipeline/MovFnAggregatorTests.java

```diff
@@ -69,14 +71,44 @@ public class MovFnAggrgatorTests extends AggregatorTestCase {
 
     @Override
     protected ScriptService getMockScriptService() {
-        MockScriptEngine scriptEngine = new MockScriptEngine(
-            MockScriptEngine.NAME,
-            Collections.singletonMap("test", script -> MovingFunctions.max((double[]) script.get("_values"))),
-            Collections.emptyMap()
-        );
+        ScriptEngine scriptEngine = new ScriptEngine() {
+            @Override
+            public String getType() {
+                return "test";
+            }
+
+            @Override
+            public <FactoryType> FactoryType compile(
+                String name,
+                String code,
+                ScriptContext<FactoryType> context,
+                Map<String, String> params
+            ) {
+                if (getSupportedContexts().contains(context) == false) {
+                    return null;
+                }
+                MovingFunctionScript.Factory factory = () -> new MovingFunctionScript() {
+                    @Override
+                    public double execute(Map<String, Object> params, double[] values) {
+                        return MovingFunctions.max(values);
+                    }
+                };
+                return context.factoryClazz.cast(factory);
+            }
+
+            @Override
+            public Set<ScriptContext<?>> getSupportedContexts() {
+                return Set.of(MovingFunctionScript.CONTEXT);
+            }
+        };
```

Comment:
This is required because we `MockScriptEngine` can no longer provide a default implementation. So far as I can tell that implementation was only used for this test so I'm kind of happy to isolate it here anyway.